### ISSUE TITLE
Honor options (jspb.message_id) in protoc for js-generated code

### ISSUE THREAD

**Issue by danbeam** [protocolbuffers/protobuf#7543]
**What language does this apply to?**
generated JS

**Describe the problem you are trying to solve.**
i'm trying to dynamically pack/unpack an `google.protobuf.Any` in JS.  in order to pack/unpack you currently need to essentially duplicate the type name of a proto as a string (assuming you basically want to reflect the same name as the type when packing).

**Describe the solution you'd like**
a way I've done this in the past was by using `options (jspb.message_id) = "TypeName"` in the .proto file itself.  then, in the generated JS, `referenceToProtoInstance.getJspbMessageId()` will return (I believe) the namespace + the given message_id.

in fact, the code is already there in js_generator.cc to do this, it's just that `GetMessageId()` returns empty string rather than looking for this option.
https://github.com/protocolbuffers/protobuf/blob/61301f01552dd84d744a05c88af95833c600a1a7/src/google/protobuf/compiler/js/js_generator.cc#L2077

additionally, this constant is in an .h file (but unused, AFAICT).
https://github.com/protocolbuffers/protobuf/blob/60b66a119d17f0a2a595a231bea87cd4f4cf2689/src/google/protobuf/util/internal/constants.h#L93

**Describe alternatives you've considered**
(none as of yet)

**Additional context**
I believe this to be a google-internal feature that just isn't publicly available, so the effort to do this sounds minimal.  ex:
https://github.com/google/rejoiner/blob/master/rejoiner/src/test/java/com/google/api/graphql/grpc/SchemaToProtoTest.java#L47

**Comment by shaod2** [comment]
Yeah I believe this option is only internally used. Do you already have any workarounds?

**Comment by danbeam** [comment]
Pretty sure I'd have to fork and compile protoc to work around, which I would prefer not to do.
